app/utils/calculate-burn-height.ts

```diff
@@ -0,0 +1,15 @@
+import BigNumber from 'bignumber.js';
+
+interface CalcBurnHeightBlockFromCyclesArgs {
+  cycles: number;
+  rewardCycleLength: number;
+  currentCycleId: number;
+  genesisBurnBlockHeight: number;
+}
+export function calculateUntilBurnHeightBlockFromCycles(args: CalcBurnHeightBlockFromCyclesArgs) {
+  const { cycles, rewardCycleLength, genesisBurnBlockHeight, currentCycleId } = args;
+  return new BigNumber(genesisBurnBlockHeight)
+    .plus(currentCycleId + 1 * rewardCycleLength)
```

Comment:
Oh man, I was so puzzled for a while, this is the issue! Order of operations are wrong, here.

```suggestion
    .plus((currentCycleId + 1) * rewardCycleLength)
```

Reply:
🤦🏼  thanks @hstove this would've been a big issue.

Updated tests an added an extra assertion the result should always be divisible by 50 (given reward cycle lengths used)